### PR TITLE
MCR-2481 split access key rest methods for objects and derivates

### PR DESCRIPTION
**Pull request by golsch**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2481).

### CODE REVIEW

File: mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKeyHelper.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import org.mycore.mcr.acl.accesskey.MCRAccessKeyManager;
+import org.mycore.mcr.acl.accesskey.MCRAccessKeyTransformer;
+import org.mycore.mcr.acl.accesskey.exception.MCRAccessKeyNotFoundException;
+import org.mycore.mcr.acl.accesskey.model.MCRAccessKey;
+import org.mycore.mcr.acl.accesskey.restapi.v2.model.MCRAccessKeyInformation;
+import org.mycore.restapi.v2.MCRErrorResponse;
+
+public class MCRRestAccessKeyHelper {
+
+    public static final String SECRET = "secret";
```
Comment (rsteph-de) [inline mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKeyHelper.java]: What is this for and how secret is this secret? -> Javadoc

**Review by rsteph-de — CHANGES_REQUESTED** [review]
The Java-Part looks O.K. Please ask @sebhofmann to review the Typescript code.

File: mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKeyHelper.java (hunk 1 above)
Comment (golsch) [reply mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKeyHelper.java]: It should be used as a uniform placeholder for secret between path and path param of MCRRestObject or MCRRestDerivate methods. It has basically not much to do with the class, but for a utils class it is too less.

Comment (rsteph-de) [reply mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKeyHelper.java]: I would change the name into something like PATH_PARAM_SECRET="secret".
At least a Javadoc comment would helpful.

Comment (golsch) [reply mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestAccessKeyHelper.java]: Ok, I changed the name to PARAM_SECRET, similar to the names in MCRRestAuthorizationFilter, and added a comment. It is difficult to write a comment about a secret that is not a secret at all. LOL

File: mycore-acl/src/main/vue/access-key-manager/src/App.vue [hunk 2]
```diff
@@ -314,7 +314,7 @@ export default class AccessKeyManager extends Vue {
       console.error('webApplicationBaseURL is not set');
       return;
     }
-    if (objectID == null) {
+    if ((objectID == null) || (derivateID != null && objectID == null)) {
```
Comment (rsteph-de) [inline mycore-acl/src/main/vue/access-key-manager/src/App.vue]: This condition looks strange: 
right condition of OR is already part of left condition

File: mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestObjectAccessKeys.java [hunk 3]
```diff
@@ -152,27 +131,23 @@ public Response getAccessKey(@PathParam(PARAM_MCRID) final MCRObjectID objectId,
...
-            @ApiResponse(responseCode = "204", description = "MCRAccessKey successfully updated"),
-            @ApiResponse(responseCode = "400", description = "Invalid request. Check type, values.",
+            @ApiResponse(responseCode = "204", description = "Access key was successfully updated"),
+            @ApiResponse(responseCode = "400", description = "Invalid ID or invalid access key",
                 content = { @Content(mediaType = MediaType.APPLICATION_JSON) }),
             @ApiResponse(responseCode = "401",
                 description = "You do not have create permission and need to authenticate first",
                 content = { @Content(mediaType = MediaType.APPLICATION_JSON) }),
-            @ApiResponse(responseCode = "404", description = "ObjectID or MCRAccessKey doesn't exists",
+            @ApiResponse(responseCode = "404", description = "Object or access key doesn't exists",
```
Comment (rsteph-de) [inline mycore-acl/src/main/java/org/mycore/mcr/acl/accesskey/restapi/v2/MCRRestObjectAccessKeys.java]: "Object or access key does not exist"
I am sure - no "s"-ending for exist / I would avoid short forms like "doesn't"

Please check all other 404-messages as well!

**Review by rsteph-de — APPROVED** [review]
looks good for me now

**Review by sebhofmann — APPROVED** [review]
typescript looks ok now

**Pull request merged**